contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -213,10 +213,14 @@ pub fn execute_close(
     let mut prop = PROPOSALS.load(deps.storage, proposal_id)?;
     if [Status::Executed, Status::Rejected, Status::Passed]
         .iter()
-        .any(|x| *x == prop.status)
+        .any(|&s| s == prop.status)
     {
         return Err(ContractError::WrongCloseStatus {});
     }
+    // Avoid closing of Passed due to expiration proposals
+    if prop.current_status(&env.block) == Status::Passed {
+        return Err(ContractError::WrongCloseStatus {});
+    }
```

Comment:
Simpler logic:

```rust
    prop.update_status(&env.block);
    if prop.status != Status::Rejected {
        return Err(ContractError::WrongCloseStatus {});
    }
```

Reply:
This doesn't preserve the current logic of expired vs. non-expired error messages. That's why I prefer the other variant.